### PR TITLE
[PM-2961] Upgrade Braintree

### PR DESCRIPTION
**Pull request by cturnbull-bitwarden**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Braintree is requiring us to upgrade to at least 5.14.0 of the SDK.
https://developer.paypal.com/braintree/docs/reference/general/server-sdk-migration-guide/dotnet

## Code changes

* **Core.csproj:** Upgrade Braintree package to version 5.19.0
* ***/packages.lock.json:** Upgrade Braintree package to version 5.19.0 and updated dependencies

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by withinfocus** [comment]
❓ Why not just go to 5.18? This is 9 months old now: https://www.nuget.org/packages/Braintree/

File: bitwarden_license/src/Commercial.Core/packages.lock.json [hunk 1]
```diff
@@ -141,8 +141,8 @@
       },
       "Braintree": {
         "type": "Transitive",
-        "resolved": "5.12.0",
-        "contentHash": "bV2tsVIvBQeKwULT4qPZUWhxSr8mFwyAAcvLDvDpCU0cMYPHzGSahha+ghUdgGMb317BqL34/Od59n2s3MkhOQ==",
+        "resolved": "5.14.0",
```
Comment (cyprain-okeke) [inline bitwarden_license/src/Commercial.Core/packages.lock.json:144]: Any reason why we show 5.14.0 instead of 5.18.0 the most recent version

**Review by cyprain-okeke — COMMENTED** [review]
It looks good. Have you been able to test it after the upgrade

**Comment by cturnbull-bitwarden** [comment]
> It looks good. Have you been able to test it after the upgrade

@cyprain-okeke yup! I was able to verify that PayPal still functions correctly

**Review by cyprain-okeke — DISMISSED** [review]
LGTM

File: util/MsSqlMigratorUtility/packages.lock.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "net6.0": {
       "Microsoft.Extensions.Logging": {
         "type": "Direct",
-        "requested": "[6.0.0, )",
```
Comment (Hinton) [inline util/MsSqlMigratorUtility/packages.lock.json:7]: It's weird that these are changing. Did you manually do something or what sofware did you use to do nuget restore?

**Review by Hinton — DISMISSED** [review]

**Review by cyprain-okeke — DISMISSED** [review]

**Review by Hinton — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Comment by withinfocus** [comment]
This is also relevant to https://github.com/bitwarden/server/pull/2975.

**Comment by cturnbull-bitwarden** [comment]
@Hinton @withinfocus @shane-melton @cyprain-okeke  This is passed through QA. I pushed a change to resolve the merge conflicts, and then re-ran `dotnet restore -f --force-evaluate --no-cache`, so the votes are reset. You'll see new changes bumping Core to `2023.9.0` as a result. Let me know if you'd rather not include those changes here

**Review by withinfocus — APPROVED** [review]
Yep the regeneration is necessary and will bump that `Core` reference, so while it's unrelated here it's no big deal. We are automating this with DevOps to happen at release (versioning) time.

**Pull request merged**